sdk/core/AzureCore/Source/Pipeline/Policies/ContentDecodePolicy.swift

```diff
@@ -218,7 +219,7 @@ public class ContentDecodePolicy: PipelineStage {
             jsonData = try JSONSerialization.data(withJSONObject: arrayObj, options: [])
         }
         guard let finalJsonData = jsonData else {
-            throw HTTPResponseError.decode("Failure decoding XML.")
+            throw AzureError.sdk("Failure decoding XML.")
```

Comment:
Let's capture the thrown error above and include it as the inner error here.

Reply:
This is the error thrown when none of the above `try` statements fail but there's still no JSON payload, so there's no inner error to wrap. See the updated logic in the `on(response:...)` for this policy.